### PR TITLE
Ticket Tag relationship fixed, Ticket tag added to api doc & Ticket tag added to dredd testing

### LINKED ISSUE #3930: Ticket-tag and tickets relationship error. Ticket tag api doc not implemented. Add ticket-tag to dredd testing
**I'm submitting a ...**  (check one with "x")
- [X] bug report
- [ ] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-orga-server

**Current behavior:**
Relationship of tags with tickets and vice versa gives error 

**Expected behavior:**
- Fix relationship
- Add ticket tag api doc
- Add ticket-tag to dredd testing

**Steps to reproduce:**

GET `v1/ticket-tags/<id>/relationships/ticket` and `v1/tickets/<id>/relationships/ticket-tags`

### PR DESCRIPTION
**Pull request by magdalenesuo**
Fixes #3930 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `nextgen` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Ticket Tag relationship fixed, Ticket tag added to api doc & Ticket tag added to dredd testing

#### Changes proposed in this pull request:

- Fix relationship with ticket and event
- Add ticket tag api doc
- Add ticket-tag to dredd testing

### CODE REVIEW

File: tests/hook_main.py [hunk 1]
```diff
@@ -870,6 +871,72 @@ def ticket_delete(transaction):
...
+
+
+@hooks.before("Ticket Tags > Ticket Tags Collection > Create Ticket Tag")
+def ticket_tag_post(transaction):
+    """
+    POST /tickets/1/ticket-tags
+    :param transaction:
+    :return:
+    """
+    with stash['app'].app_context():
```
Comment (shubham-padia) [inline tests/hook_main.py]: use event factory here :)

Comment (magdalenesuo) [inline tests/hook_main.py]: @shubham-padia Ticket Factory is already used :laughing:

**Comment by shubham-padia** [comment]
Sorry If I'm missing something :smile: , but as this is a many to many relationship there is a join table named `association` for the same. Just curious, how is the data handled for that. 
Or is the above not part of this issue, and is supposed to be handled in another issue ?

File: tests/hook_main.py (hunk 1 above)
Comment (shubham-padia) [inline tests/hook_main.py]: oh sorry, nevermind, I read tickettag factory :P

Comment (shubham-padia) [inline tests/hook_main.py]: Resolved

**Comment by magdalenesuo** [comment]
@shubham-padia this was already implemented here [#3811](https://github.com/fossasia/open-event-orga-server/pull/3811/files#diff-7c5816d7384caab73f6c257ba88236c7R67)
Basically, obj has the ticket_tag that has just been added to the db. we add its id and the ticket_id in the `association` table using the append line :)
And yes it is many to many. Ticket tag be can be relation-less with ticket too so it does not go to the association table. But after you mentioned, I realised I do not have a check if it is created via event endpoint. I will fix that.

**Comment by shubham-padia** [comment]
okay cool, thanks :)

**Comment by magdalenesuo** [comment]
@shubham-padia @SaptakS  Please review again. Resolved conflicts

**Pull request merged**